### DISCUSSION TITLE
jailmkr restart function

### DISCUSSION THREAD

**Discussion by spusuf** [Jip-Hop/jailmaker#38] [Ideas]
sometimes to test services or reload changes it is necessary to restart the jail. Running 'jlmkr stop jailname && jlmkr start jailname' results in 'the jail is already running', running jlmkr start again starts the jail sucessfully. Can we add a function to stop, wait until fully stopped then start?

Much appreciated

**Comment by Jip-Hop** [comment]
I suppose a restart command could be added which would wait for the jail to stop first similar to how this is done for the remove function: 

https://github.com/Jip-Hop/jailmaker/blob/cab79e3a245e58e552708bff9049972e91df0f55/jlmkr.py#L874-L880

Or the stop command should wait for the jail to fully stop before returning...

**Reply by spusuf** [reply]
how about both? both is good!

**Comment by Lockszmith-GH** [comment]
wouldn't `jlmkr stop jailname && sleep 5s && jlmkr start jailname` do the trick?

**Reply by Jip-Hop** [reply]
In theory a jail can take longer than 5 seconds to shutdown cleanly...

**Reply by spusuf** [reply]
some jails might cleanly stop instantly and some may take over 5s, theres no point adding an arbitrary sleep when you could just have the script check if its shut down then restart as soon as its ready.

**Comment by Jip-Hop** [comment]
I'm locking this discussion as I've opened dedicated issues to track feature requests. Please see [Restart function](https://github.com/Jip-Hop/jailmaker/issues/81).
